launcher/container_runner.go

```diff
@@ -194,43 +211,106 @@ func (r *ContainerRunner) measureContainerClaims(ctx context.Context) error {
 		}
 	}
 
-	contianerSpec, err := r.container.Spec(ctx)
+	containerSpec, err := r.container.Spec(ctx)
 	if err != nil {
 		return err
 	}
-	for _, arg := range contianerSpec.Process.Args {
+	for _, arg := range containerSpec.Process.Args {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.ArgType, EventContent: []byte(arg)}); err != nil {
 			return err
 		}
 	}
-	for _, env := range contianerSpec.Process.Env {
+	for _, env := range containerSpec.Process.Env {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.EnvVarType, EventContent: []byte(env)}); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// Run the container
-// Doesn't support container restart yet
-// Container output will always be redirected to stdio for now
-func (r *ContainerRunner) Run(ctx context.Context) error {
-	if err := r.measureContainerClaims(ctx); err != nil {
-		return fmt.Errorf("failed to measure container claims: %v", err)
+func getTTL(token []byte) (time.Duration, error) {
+	claims := &jwt.StandardClaims{}
+	_, _, err := new(jwt.Parser).ParseUnverified(string(token), claims)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse token: %w", err)
 	}
 
+	return time.Until(time.Unix(claims.ExpiresAt, 0)), nil
+}
+
+func (r *ContainerRunner) fetchAndWriteToken(ctx context.Context) error {
 	// Create hostTokenPath directory. Should come before NewTask.
 	token, err := r.attestAgent.Attest(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to retrieve attestation service token: %v", err)
 	}
 
-	// Write OIDC token for the container.
-	if err := os.MkdirAll(HostTokenPath, 0644); err != nil {
+	filepath := path.Join(HostTokenPath, attestationVerifierTokenFile)
+
+	ttl, err := getTTL(token)
```

Comment:
Added `refreshToken()`, kept `fetchAndWriteToken` in its own function for ease of testing.

The immediate goroutine seems a little complicated to set up. Don't think it's necessary since there isn't that much duplicated code here anymore.